designdocs/privacy-design.md

```diff
@@ -0,0 +1,163 @@
+
+If user consent is required, it must be obtained as part of the [XRRequestSession](https://immersive-web.github.io/webxr/#dom-xr-requestsession) algorithm before the new [XRSession](https://immersive-web.github.io/webxr/#xrsession-interface) object is created. If consent is not obtained, the user agent MUST [reject](https://www.w3.org/2001/tag/doc/promises-guide/#reject-promise) the promise with a "[SecurityError](https://heycam.github.io/webidl/#securityerror)" [DOMException](https://heycam.github.io/webidl/#idl-DOMException).
+
+```
+Editor Note: The paragraph above suggests specific changes to the XRSession algorithm language.
+```
+
+#### Providing XRViewerPose data
+The user agent must verify that all mandatory conditions are satisfied to ensure it can expose [XRViewerPose](https://immersive-web.github.io/webxr/#xrviewerpose) data in response to each call to [getViewerPose](https://immersive-web.github.io/webxr/#dom-xrframe-getviewerpose) within a given [active document](https://html.spec.whatwg.org/multipage/browsers.html#active-document).
+
+The **_mandatory conditions_** for exposing XRViewerPose data are the following:
+*   [Visibility state](https://www.w3.org/TR/page-visibility-2/#dom-visibilitystate) of the document is "visible".
+*   [Currently focused area](https://html.spec.whatwg.org/multipage/interaction.html#currently-focused-area-of-a-top-level-browsing-context) belongs to a document whose origin is [same origin-domain](https://html.spec.whatwg.org/multipage/origin.html#same-origin-domain) with the origin of the given [active document](https://html.spec.whatwg.org/multipage/browsers.html#active-document).
+*   The document is [allowed to use](https://wicg.github.io/feature-policy/#allowed-to-use) all the [policy-controlled features](https://wicg.github.io/feature-policy/#policy-controlled-feature) associated with the given [sensor types](https://www.w3.org/TR/2018/CR-generic-sensor-20180320/#sensor-type) of sensors used to generate [XRViewerPose](https://immersive-web.github.io/webxr/#xrviewerpose) data.
+*   If the reference space is of type [bounded](https://immersive-web.github.io/webxr/#dom-xrreferencespacetype-bounded), [floor-level](https://immersive-web.github.io/webxr/#dom-xrreferencespacetype-floor-level) or [eye-level](https://immersive-web.github.io/webxr/#dom-xrreferencespacetype-eye-level), the [XRViewerPose](https://immersive-web.github.io/webxr/#xrviewerpose) data must be limited to a region approximately the size of a large room, to prevent such data from being used to [determine the user’s real-world location](https://github.com/immersive-web/privacy-and-security/blob/master/POSE-AND-ENVIRONMENT.md#user-location). Specific bounds are at the discretion of the user agent, but 30m x 30m is suggested as a reasonable limit. These bounds may be affected by previously created reference spaces.
```

Comment:
An earlier revision did indeed break this out, but I had feedback (I believe from @toji) that different bounds seemed arbitrary and reflective of the spec design, instead of the underlying privacy threat and mitigation. In practice room scale bounds are sufficient to mitigate the privacy threat for all reference space types.

There is also a complicating factor mentioned later in the doc that makes it desirable to have a single bounds: If you create multiple reference spaces in the same session, they should all have the same maximum bounds, to prevent the creation of a series of references spaces that can be compared to indirectly get the effect of an unbounded space. That complication suggests that it'd be desirable for a single size recommendation, allowing the user agent to establish the maximum bounded region with the first reference space, instead of potentially having to 'grow' it depending upon which reference space was created in which order.

Having said that, I don't have a strong opinion... as long as the bounds mitigate the privacy threat, there is some flexibility in approach here. @toji curious to hear your thoughts.

Reply:
As one more note, a benefit of more restrictive bounds for `local/local-floor` is that it helps mitigate the threat of user fingerprinting based upon gait and height (since a more limited position makes this more difficult).

Reply:
>allowing the user agent to establish the maximum bounded region with the first reference space, instead of potentially having to 'grow' it depending upon which reference space was created in which order.

This is a great clarification!  Can we get this captured in the design doc?

Reply:
There's one other way to think about this:
If we're saying that local/local-floor < bounded-floor < unbounded, then perhaps that should be the permission being requested at session creation time.  I'd originally been hesitating about this approach because I was concerned about the implications on adding new reference spaces in the future (such as diorama), but I think we can choose to map new reference spaces to share a "safe" bounds as we design them.  For example we might say [local/local-floor/diorama] < bounded < unbounded.

My biggest concern about this is that it will encourage developers to ask for the maximum bounds "just in case".